libbeat/cmd/instance/beat.go

```diff
@@ -979,11 +982,104 @@ func (b *Beat) makeOutputFactory(
 	}
 }
 
+func (b *Beat) reloadOutputOnCertChange(cfg config.Namespace) error {
+	// TODO (Tiago): find a proper name for this logger
+	logger := logp.L().Named("ssl.cert.reloader")
+	// Here the output is created and we have acces to the Beat struct (with the manager)
+	// as a workaround we can unpack the new settings and trigger the reload-watcher from here
+
+	// We get an output config, so we extract the 'SSL' bit from it
+	rawTLSCfg, err := cfg.Config().Child("ssl", -1)
+	if err != nil {
+		e, ok := err.(ucfg.Error)
+		if ok {
+			if errors.Is(e.Reason(), ucfg.ErrMissing) {
+				// if the output configuration does not contain a `ssl` section
+				// do nothing and return no error
+				return nil
+			}
+		}
+		return fmt.Errorf("could not extract the 'ssl' section of the output config: %w", err)
+	}
+
+	extendedTLSCfg := struct {
+		tlscommon.Config `config:",inline" yaml:",inline"`
+		Reload           cfgfile.Reload `config:"exit_on_cert_change" yaml:"exit_on_cert_change"`
+	}{
+		Reload: cfgfile.Reload{
+			Enabled: false,
+			Period:  time.Minute,
+		},
+	}
+
+	if err := rawTLSCfg.Unpack(&extendedTLSCfg); err != nil {
+		return fmt.Errorf("unpacking 'ssl' config: %w", err)
+	}
+
+	if !extendedTLSCfg.Reload.Enabled {
+		return nil
+	}
+	logger.Debug("exit on CA certs change enabled")
+
+	possibleFilesToWatch := append(
+		extendedTLSCfg.CAs,
+		extendedTLSCfg.Certificate.Certificate,
+		extendedTLSCfg.Certificate.Key,
+	)
+
+	filesToWatch := []string{}
+	for _, f := range possibleFilesToWatch {
+		if f == "" {
+			continue
+		}
+		if tlscommon.IsPEMString(f) {
+			// That's an embedded cert, we're only interested in files
+			continue
+		}
+
+		logger.Debugf("watching '%s' for changes", f)
+		filesToWatch = append(filesToWatch, f)
+	}
+
+	watcher := filewatcher.New(filesToWatch...)
+	// Ignore the first scan as it will always return
+	// true for files changed. The output has not been
+	// started yet, so even if the files have changed since
+	// the Beat started, they don't need to be reloaded
+	watcher.Scan()
+
+	// Watch for file changes while the Beat is alive
+	go func() {
+		ticker := time.Tick(extendedTLSCfg.Reload.Period)
```

Comment:
```suggestion
		ticker := time.NewTicker(extendedTLSCfg.Reload.Period)
		defer ticker.Stop()
```

This will fix the linter.

Reply:
But why? There is no benefit from using `time.Tick` over `time.NewTicker`, it's worth to change to fix the linter, it's the same effort as adding the `nolint` directive.

Reply:
Using `defer` on a function that never ends has no effect. Essentially it's dead code. It might even lead the reader to believe the deferred code will eventually run.

That is the indented purpose of `time.Tick`, from the docs:
```
// Tick is a convenience wrapper for NewTicker providing access to the ticking
// channel only. While Tick is useful for clients that have no need to shut down
// the Ticker (...)
```

If the point here is to stop the linter errors, the comment fixes the linter without adding dead code.

Reply:
If you think it's better to write a `nolint` directive instead of a `defer` statement, go ahead, it would not block the review. Either way we need to fix the linter here.